server/public/plugin/api.go

```diff
@@ -881,7 +881,7 @@ type API interface {
 	//
 	// @tag Plugin
 	// Minimum server version: 5.6
-	RemovePlugin(id string) *model.AppError
+	RemovePlugin(id string, r *http.Request) *model.AppError
```

Comment:
@willypuzzle, I can't see a reason for why `RemovePlugin` and `InstallPlugin` require a `*http.Request`. These are methods that are designed to be used outside of an HTTP request, so something seems amiss here.

Reply:
yes, I could not pass the `http.Request` in these calls but this implies that the context passed in `OnPluginStatusesChanged` will bell always null, when such methods are called.

Reply:
@lieut-data so, what should I do? passing nothing and allowing null context in any case?

Reply:
I'd have to double check, but I think the pattern would be to create an empty context. Are there some fields that we can fill in for the context while leaving the others empty (because they didn't come from an `http.Request` in the first place?)

Reply:
Unfortunately not, are all http based information https://github.com/mattermost/mattermost/blob/master/server/channels/app/context.go#L17-L25

Reply:
Then just an empty context should suffice: we do this elsewhere with hooks, for example:

https://github.com/mattermost/mattermost/blob/369b9fa96b4663de470b7c210da34caca7992783/server/channels/app/cluster_handlers.go#L51-L54